### DISCUSSION TITLE
How to use openmdx 1st level cache

### DISCUSSION THREAD

**Discussion by nyemiks** [openmdx/openmdx#37] [Q&A] [answered]
I am working on an application where I want some data fetched from database to be cached.
So that subsequent queries do not have to go to the database. This data does not change frequently. 
It is mostly configuration data that is used while the app is running.

From a coding point of view How can I make sure this data is not being fetched over and over when it is used ?

Let me share some code snippets in use

 pm = ConnectionManager.getPersistenceManager(userId);
        
            SegmentUtil segmentUtil = new SegmentUtil();     
            Segment segment = segmentUtil.getExtensionSegment(pm);
            
           // configdata is a class that holds configuration info that is used over and over in the application
      
          ConfigDataQuery configQuery = (ConfigDataQuery)pm.newQuery(ConfigData.class);
        configQuery.propertyKey().equalTo(propertyKey);

      List<ConfigData> propertyList = segment.getConfigData(configQuery);
      ConfigData config = propertyList.get(0);

    // use configdata in application logic 
		 .............

     // then close pm here
        pm.close();

  // method ends here

Is the above usage correct ? will subsequent calls to getConfigData return the cached object ?

**Comment by DlRTY-HARRY** [comment, marked as answer]
Yes, this usage is correct: JDO instances which are referenced, e.g. by `config`, are excluded from eviction.

**Reply by nyemiks** [reply]
thanks Harry.
